src/interpreter.rs

```diff
@@ -120,3 +129,1207 @@ bitflags::bitflags! {
+                            let mut ikey = i;
+                            i += keys_count;
+                            if stack.size() < i.into() {
+                                return set_error(ScriptError::InvalidStackOperation);
+                            }
+
+                            let mut sigs_count =
+                                u8::try_from(ScriptNum::new(stack.top(-isize::from(i))?, require_minimal, None)?)
+                                  .map_err(|_| ScriptError::SigCount)?;
+                            if sigs_count > keys_count {
+                                return set_error(ScriptError::SigCount);
+                            };
+                            i += 1;
+                            let mut isig = i;
+                            i += sigs_count;
+                            if stack.size() < i.into() {
+                                return set_error(ScriptError::InvalidStackOperation);
+                            };
+
+                            let mut success = true;
+                            while success && sigs_count > 0 {
+                                let vch_sig: &ValType = stack.top(-isize::from(isig))?;
+                                let vch_pub_key: &ValType = stack.top(-isize::from(ikey))?;
+
+                                // Note how this makes the exact order of pubkey/signature evaluation
+                                // distinguishable by CHECKMULTISIG NOT if the STRICTENC flag is set.
+                                // See the script_(in)valid tests for details.
+                                if !check_signature_encoding(vch_sig, flags)? {
+                                    // serror is set
+                                    return Ok(false);
+                                };
+                                check_pub_key_encoding(vch_pub_key, flags)?;
+
+                                // Check signature
+                                let ok: bool = checker.check_sig(vch_sig, vch_pub_key, script);
+
+                                if ok {
+                                    isig += 1;
+                                    sigs_count -= 1;
+                                }
+                                ikey += 1;
+                                keys_count -= 1;
+
+                                // If there are more signatures left than keys left,
+                                // then too many signatures have failed. Exit early,
+                                // without checking any further signatures.
+                                if sigs_count > keys_count {
+                                    success = false;
+                                };
+                            }
+
+                            // Clean up stack of actual arguments
+                            while {
+                                let res = i > 1;
+                                i -= 1;
+                                res
+                            } {
```

Comment:
Since `i` is not used after this, it is equivalent to
```suggestion
                            for _ in 1..i {
```